src/XMakeTasks/Microsoft.Common.CurrentVersion.targets

```diff
@@ -3166,6 +3167,22 @@ Copyright (C) Microsoft Corporation. All rights reserved.
 
   <!--
     ============================================================
+                                        _GenerateCompilerDependencyFile
+
+    Generate file with a list of compile source files. This is used to track
+    incremental build between build execution specifically around addition and
+    deletions of file globs (e.g. <Compile Include="**\*.cs" />).
+    ============================================================
+    -->
+  <Target Name="_GenerateCompilerDependencyFile" BeforeTargets="CoreCompile">
+    <WriteLinesToFile Lines="@(Compile->'%(FullPath)')" File="$(IntermediateOutputPath)Compile.deps" Overwrite="True" WriteOnlyWhenDifferent="True" />
```

Comment:
Definitely also references (via `ReferencePath` and `DependsOnTargets="ResolveAssemblyReferences"`, probably?). Other properties and items _should_ be caught by the `Inputs="$(MSBuildAllProjects)"` [in `Microsoft.CSharp.Core.targets`](https://github.com/dotnet/roslyn/blob/98fb1f2a1a784e7724a3f043af342484f323703c/src/Compilers/Core/MSBuildTask/Microsoft.CSharp.Core.targets#L4). That would be more reliable with #1299 (as is, it might not catch some user-authored files) but is . . . ok now.